drivers/gpu/drm/lima/lima_device.c

```diff
@@ -78,6 +79,31 @@ static void lima_clk_fini(struct lima_device *dev)
 	clk_disable_unprepare(dev->clk_bus);
 }
 
+static int lima_regulator_init(struct lima_device *dev)
+{
+	int ret;
+	dev->reg = devm_regulator_get_optional(dev->dev, "mali");
+	if (IS_ERR(dev->reg)) {
+		dev_err(dev->dev, "failed to get regulator: %ld\n", PTR_ERR(dev->reg));
```

Comment:
On sunxi platforms I don't see this regulator, so we'd better dev_err if error != ENODEV.